src/aws-cpp-sdk-core/source/auth/GeneralHTTPCredentialsProvider.cpp

```diff
@@ -0,0 +1,261 @@
+
+
+bool IsAllowedIp(const Aws::String& authority)
+{
+    // address is an ECS / EKS container host
+    if (authority == GeneralHTTPCredentialsProvider::AWS_ECS_CONTAINER_HOST) // ECS container host
+    {
+        return true;
+    } else if (authority == GeneralHTTPCredentialsProvider::AWS_EKS_CONTAINER_HOST ||
+               authority == GeneralHTTPCredentialsProvider::AWS_EKS_CONTAINER_HOST_IPV6) { // EKS container host
+        return true;
+    }
+
+    // address is within the loop back
+    // IPv4
+    if (authority.rfind(Aws::String("127.0.0."), 0) == 0 && authority.size() > 8 && authority.size() <= 11)
+    {
+        Aws::String octet = authority.substr(8);
+        if ((octet.find_first_not_of("0123456789") != Aws::String::npos) ||
+            (Aws::Utils::StringUtils::ConvertToInt32(octet.c_str()) > 255)) {
+            AWS_LOGSTREAM_WARN(GEN_HTTP_LOG_TAG, "Can't use General HTTP Provider: AWS_CONTAINER_CREDENTIALS_FULL_URI ip address is malformed: " << authority);
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    // IPv6
+    if (authority == "::1" || authority == "0:0:0:0:0:0:0:1" || authority == "[::1]" || authority == "[0:0:0:0:0:0:0:1]")
+    {
+        return true;
+    }
+
+    return false;
+}
+
+bool ShouldCreateGeneralHTTPProvider(const Aws::String& relativeUri, const Aws::String& absoluteUri, const Aws::String authToken)
+{
+    if (authToken.find("\r\n") != Aws::String::npos)
+    {
+        AWS_LOGSTREAM_WARN(GEN_HTTP_LOG_TAG, "Can't use General HTTP Provider: AWS_CONTAINER_AUTHORIZATION_TOKEN env value contains invalid characters (\\r\\n)");
+        return false;
+    }
+
+    if (!relativeUri.empty())
+    {
+        // The provider MAY choose to assert syntactical validity of the resulting URI
+        // perform very basic check here
+        if (relativeUri[0] != '/') {
+          AWS_LOGSTREAM_WARN(GEN_HTTP_LOG_TAG, "Can't use General HTTP Provider: AWS_CONTAINER_CREDENTIALS_RELATIVE_URI does not begin with /");
+          return false;
+        } else {
+          // full URI is not used in case of a relative one present
+          return true;
+        }
+    }
+
+    if (!absoluteUri.empty())
+    {
+        // If the resolved URI’s scheme is HTTPS, its hostname may be used in the request
+        if (Aws::Utils::StringUtils::ToLower(absoluteUri.c_str()).rfind(Aws::String("https://"), 0) == 0) // if starts_with
+        {
+            return true;
+        }
+
+        Aws::Http::URI absUri(absoluteUri);
+        const Aws::String& authority = absUri.GetAuthority();
+
+        // Otherwise, implementations MUST fail to resolve when the URI hostname does not satisfy any of the following conditions
+        if (IsAllowedIp(authority))
+        {
+            return true;
+        }
+
+        Aws::Crt::Io::HostResolver* pHostResolver = Aws::Crt::ApiHandle::GetOrCreateStaticDefaultHostResolver();
+        if (pHostResolver)
```

Comment:
should we create a configuration option to disable DNS lookup incase a existing customer is using this and DNS lookup break their use case.

Reply:
added option to override the restriction to use HTTPs / pre-allowed addresses only.